### ISSUE TITLE
Outdated version of @honeybadger-io/js 3.2.8 => 4.0.2

### ISSUE THREAD

**Issue by JustFly1984** [honeybadger-io/honeybadger-react#326]
updating both @honeybadger-io/js and @honeybadger-io/react to latest version produces dependency conflict in yarn.lock

```
"@honeybadger-io/js@4.0.2", "@honeybadger-io/js@^3.2.8":
  version "4.0.2"
  resolved "https://registry.yarnpkg.com/@honeybadger-io/js/-/js-4.0.2.tgz#80c40eeb06f7379f43e36926a8490cfb5bc89096"
  integrity sha512-kK8UMc/2uA5WPFsn+Tj2ZTcwlVtctjPTogXk8QGCNirTKQV+5TGMm+dhpHP8R2Nj47oEQI3B/gnU9+U/gFEqzw==
  dependencies:
    stacktrace-parser "^0.1.10"

"@honeybadger-io/react@2.0.1":
  version "2.0.1"
  resolved "https://registry.yarnpkg.com/@honeybadger-io/react/-/react-2.0.1.tgz#50651ebd93fbb0daa2688d14d934301348cf62e8"
  integrity sha512-7Sr0ExI+s4CDYXPuZf9nUoy6KUofzyrwH09ZcjUVELySGrSpaBEDJplGzZ3tIdaV/oYkltWfyRaC+rnghsiH6g==
  dependencies:
    "@honeybadger-io/js" "^3.2.8"
 ```

**Comment by subzero10** [comment]
Hey @JustFly1984, thanks for submitting an issue!

Are you saying that you have both Honeybadger packages in your dependencies array?
Like:
```
"dependencies": {
    "@honeybadger-io/js": "^4.0.2"
    "@honeybadger-io/react": "^3.2.9"
    ...
  }
```
If yes, you shouldn't need to do that. `@honeybadger-io/js` is a dependency of `@honeybadger-io/react` and it should always install the correct version.

FYI, I'm working on bringing the latest version of `@honeybadger-io/js` to `@honeybadger-io/react`.

**Comment by JustFly1984** [comment]
We have nextjs app, with graphql @apollo/client which has both server and frontend part.

The only import we are using from `/react` is 	`import { HoneybadgerErrorBoundary } from '@honeybadger-io/react';`

All other imports in our app is dode from `/js`

`import type { Noticeable } from '@honeybadger-io/js/dist/browser/types/core/types';`

```
/* Using alpha version of universal HB because hb server doesn't work since next 9.6+ */
import Honeybadger from '@honeybadger-io/js';
import type Client from '@honeybadger-io/js/dist/server/types/core/client.js';

// TODO: use Partial<> or fill in missing config settings
const config: Partial<Client['config']> = {
  apiKey: process.env['HONEYBADGER_API_KEY'] || '',
  environment: process.env['HONEYBADGER_ENV'] || '',
  revision: process.env['HONEYBADGER_REVISION'] || '',
  // onerror: false,
  // Docs outdated https://github.com/honeybadger-io/honeybadger-js/blob/master/src/browser.ts#L133
  enableUncaught: false,
};

const honeybadger = Honeybadger.configure(config);

export default honeybadger;
```

**Comment by subzero10** [comment]
No, I think this is good. Can you please post your dependencies in your package.json file?
I'd like to know if you are installing both `@honeybadger-io/js` and `@honeybadger-io/react` or just `@honeybadger-io/react`.

Thanks!

**Comment by subzero10** [comment]
> I'm installing both, cos else eslint warns about missing dependency. It is better to set `@honeybadger-io/js` as peer dependency in `@honeybadger-io/react`

Hmm, I see. It may be a good idea to have the core package as peer dependency.
@joshuap What do you think? I can see that others are doing it also ([react](https://github.com/facebook/react/blob/main/packages/react-dom/package.json)). It will be a breaking change though, so we'll have to release a major version update.

For now, I have fixed the conflicts and react@18 warning with this [PR](https://github.com/honeybadger-io/honeybadger-react/pull/322) and this [PR](https://github.com/honeybadger-io/honeybadger-react/pull/329). Last one should be merged and released soon.

**Comment by joshuap** [comment]
> > I'm installing both, cos else eslint warns about missing dependency. It is better to set `@honeybadger-io/js` as peer dependency in `@honeybadger-io/react`
> 
> Hmm, I see. It may be a good idea to have the core package as peer dependency. @joshuap What do you think? I can see that others are doing it also ([react](https://github.com/facebook/react/blob/main/packages/react-dom/package.json)). It will be a breaking change though, so we'll have to release a major version update.
> 
> For now, I have fixed the conflicts and react@18 warning with this [PR](https://github.com/honeybadger-io/honeybadger-react/pull/322) and this [PR](https://github.com/honeybadger-io/honeybadger-react/pull/329). Last one should be merged soon and we'll release a new version right after.

I'm OK with doing a major release if this is the right way to do it, and it will make the developer experience smoother long-term.

**Comment by subzero10** [comment]
Sounds good. I have created #332 to look into this further and take action.

**Comment by subzero10** [comment]
This should be fixed with v2.1.0.
